### DISCUSSION TITLE
MonitorControl v4.0.2

### DISCUSSION THREAD

**Discussion by JoniVR** [MonitorControl/MonitorControl#798] [Release Discussion]
This is a service release with some minor changes.

### Improvements

- Settings are now saved for virtual displays (Sidecar, DisplayLink etc.) as well.
- Shorter wake wait times, removed "locked control" OSD sign.

### Fixes

- Menuless operation's Quit button location now matches the corresponding info text.

### Notes

- If you discover any problem, feel free to open an issue!
- If you want to help translating, see #637
- Thanks to everyone who helps make MonitorControl better!

<hr /><em>This discussion was created from the release <a href='https://github.com/MonitorControl/MonitorControl/releases/tag/v4.0.2'>MonitorControl v4.0.2</a>.</em>

**Comment by hagncola** [comment]
Is it possible to get config for "Shorter wake wait times", with 4.02 release sometimes after wake from sleep the monitors (two display setup) take to long for wake and the old setting does not apply.

**Reply by waydabber** [reply]
hi @hagncola - does it work well with 4.0.1? We could add a setting for this.

**Reply by hagncola** [reply]
Hi, it work´s in the meanwhile. I use USB-C to HDMI cable instead of USB-C to USB-C for the second monitor (Mac Mini M1)
I found only one issue (i don´t know if it´s related to last version). If i use the the built in autostart option, after restart the "old brightness" does not apply -- always 100 %. If i use the autostart option  from monterey  and disbale the built in autostart it works.

**Reply by waydabber** [reply]
Interesting. Maybe it is related to how quickly the display gets responsive maybe? You could try this with v4.0.1 as well.

**Reply by hagncola** [reply]
As i wrote, now all is working fine with 4.02. Never touch running system :)

**Reply by cristhianaguilargiese** [reply]
Works amazing with a Spectre monitor, great work!

**Comment by aganiban** [comment]
Recently installed the latest release and my original brightness level of my LED Cinema Display dropped, I tried to increase the brightness but it wont get any brighter than the original

**Reply by waydabber** [reply]
Hi @aganiban , did you try setting 255 to Brightenss DDC Max under Displays/Show advanced...? This is needed for LED Cinema Displays.

**Reply by dyasol** [reply]
I have the same issue.  My Apple Thunderbolt Display 27" max brightness is at probably around 25% with brightness slider at max.  I have DDC max populated with 255.  

I tried MonitorControl 4.0.2 because native brightness control was no longer available under Catalina on my 2012 Macbook Pro.  My monitor was previously stuck at max brightness.  After MonitorControl, now it's stuck at about 25%.

**Reply by waydabber** [reply]
Hi @dyasol, can you change the brightness using ddcctl or m1ddc?

**Reply by dyasol** [reply]
DDCctl did not work for me.  In MonitorControl, I disabled DDC hardware control, then I used Lunar to reset the brightness to max.  Now, using software control only, brightness adjustment with MonitorControl works properly.  I cannot enable DDC hardware control at all.  As soon as I do, the display dims again requiring me to use Lunar to reset it.

On a side note, when using Lunar to reset the max brightness, I have to immediately exit Lunar after resetting or else the display dims again.

**Reply by waydabber** [reply]
Hey @alin23 - it would be interesting to figure out what is happening with these displays as this issue comes up from time to time. It seems like both Lunar and MonitorControl has issues (while MC seems to work with other TB and Cinema displays when max DDC brightness limit is set to 255).

**Reply by waydabber** [reply]
Also: https://github.com/MonitorControl/MonitorControl/discussions/739

**Reply by alin23** [reply]
Thunderbolt displays and newer LED Cinema displays have extended brightness range.

So, while 99% of the monitors can receive brightness values between 0 and 100, Thunderbolt and Cinema can receive values between 0 and 255.

Lunar already sets the upper DDC limit to 255 for LED Cinema. I didn't do this for Thunderbolt because I thought everyone will just use the Apple Native protocol, I don't understand why you would use DDC for a Thunderbolt display.

If you really want to keep using DDC, you can raise the max brightness in Lunar as well:
<img width="1041" alt="image" src="https://user-images.githubusercontent.com/3819725/146194413-9768f27a-fce2-4155-9589-51033aa9097a.png">

**Reply by dyasol** [reply]
I am using MC for my Thunderbolt display because since High Sierra, Apple seems to have disabled the FaceTime cam, audio and native brightness control on the TB display.  Due to Apple's forced obsolescence, my TB display has been reduced to a basic monitor.  I have only recently stumbled upon MonitorControl and I'm glad to have found it because at least I now have brightness control back.

**Reply by waydabber** [reply]
@dyasol - does hardware brightness control work with Lunar when you force set the ddc max to 255? If so and it does not work with MC using the same setting (DDC brightness max set to 255) then somehow Apple TB display is not fully compatible with the way MC tries to control it. LED Cinema Display users tend to report success with MC, for Thunderbolt display users the results are mixed so far using DDC.

**Reply by alin23** [reply]
@dyasol The native brightness control still works for Thunderbolt systems on all macOS systems. I have data about working Apple Native protocol on these displays for Catalina, Big Sur and Monterey, albeit only on newer Mac devices.

I'm not sure why it isn't working for you. Maybe the new macOS versions dropped DisplayServices support for older Macbooks.

I've compiled a Lunar beta here with the 255 limit automatically applied for detected Thunderbolt displays: https://files.lunar.fyi/Lunar-5.1.2-beta14.zip

You might need to do a Full Reset in Lunar if the 100 limit is stored in settings.
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/3819725/146203514-3547a1ab-dc86-437b-8ab4-bab2a9a4912c.png">

**Reply by dyasol** [reply]
> @dyasol - does hardware brightness control work with Lunar when you force set the ddc max to 255? If so and it does not work with MC using the same setting (DDC brightness max set to 255) then somehow Apple TB display is not fully compatible with the way MC tries to control it. LED Cinema Display users tend to report success with MC, for Thunderbolt display users the results are mixed so far using DDC.

Forcing ddc max to 255 does not seem to work in DDCctrl, MC or Lunar.  Only doing a Brightness/contrast or full reset in Lunar seems to accomplish that.

**Reply by dyasol** [reply]
> @dyasol The native brightness control still works for Thunderbolt systems on all macOS systems. I have data about working Apple Native protocol on these displays for Catalina, Big Sur and Monterey, albeit only on newer Mac devices.
> 
> I'm not sure why it isn't working for you. Maybe the new macOS versions dropped DisplayServices support for older Macbooks.
> 
> I've compiled a Lunar beta here with the 255 limit automatically applied for detected Thunderbolt displays: https://files.lunar.fyi/Lunar-5.1.2-beta14.zip
> 
> You might need to do a Full Reset in Lunar if the 100 limit is stored in settings. <img alt="image" width="1062" src="https://user-images.githubusercontent.com/3819725/146203514-3547a1ab-dc86-437b-8ab4-bab2a9a4912c.png">

I am running Catalina on a 2012 MBP with my TB display, so I agree, they probably dropped DisplayServices support on old hardware.

**Reply by LBDinCA** [reply]
I was so excited but unfortunately MC did not work for me either. After installation, it decreased my screen brightness even further. I have a Cinema Display with MacBook Air (Apple M1 chip) running Monterey.

**Comment by pyroranking** [comment]
Hi. The sound control keys does not work (nor slider) with **MacBook Pro M1-Pro** in a **HP 27ea** monitor screen... (macOS **Monterey 12.1** + **MonitorControl 4.0.2** Compilation 6965). The sound symbols (increase/decrease and mute) appear on the screen but they do not control the external speakers of the monitor.
The brightness keys works fine (thank you!).
What could be the solution? Thank you in advance!

**Comment by ajaffarali** [comment]
The volume controls for the Samsung G9 don't work. I have it connected to my M1 Mac Mini using the Thunderbolt to DisplayPort cable. Brightness works so it is sending commands to the monitor but volume levels aren't changing

**Reply by waydabber** [reply]
This is probably a display firmware problem.

**Reply by ajaffarali** [reply]
So I connected a Windows PC (DP to DP cable) and that can control the volume using the volume keys on the keyboard without any app installed. Not sure if that bit of info helps but thought I'd mention. No luck with the Mac though.

**Reply by alin23** [reply]
@ajaffarali Windows controls a software volume (dimming the volume in software before it reaches the monitor).

DDC-based apps like MonitorControl and [Lunar](https://lunar.fyi/) change the monitor volume in hardware (the same value that you can see in the monitor settings using its physical buttons).

If you're trying to achieve what Windows is doing, [SoundSource](https://rogueamoeba.com/soundsource/) can do that for macOS

**Reply by ajaffarali** [reply]
Gotcha. Since MonitorControl (and Lunar for that matter) don't work for volume levels on my Samsung G9, I'll use SoundSource that does the trick and is good enough for me. 

Appreciate your help.

**Comment by leiiiooo** [comment]
Save my life, really, thanks.
